lib/blocs/channels_cubit/channels_cubit.dart

```diff
@@ -305,26 +314,28 @@ abstract class BaseChannelsCubit extends Cubit<ChannelsState> {
           emit(newState);
           break;
         case ResourceAction.deleted:
-          // fill up required fields with dummy data
-          change.resource['name'] = '';
-          change.resource['permissions'] = const [];
-
-          Logger().w('DELETED: ${change.resource}');
-          final deleted =
-              Channel.fromJson(json: change.resource, jsonify: false);
-
-          _repository.delete(
-            channel: deleted,
-            syncRemote: false,
-          );
-
-          channels.removeWhere((c) => c.id == deleted.id);
-
-          emit(ChannelsLoadedSuccess(
-            channels: channels,
-            hash: channels.fold(0, (acc, c) => acc + c.hash),
-            selected: selected,
-          ));
+          if (change.resource['id'] != Globals.instance.channelId) {
```

Comment:
hmm, can you tell me why we need this condition?

Reply:
We need here to avoid ChannelFromJson serialisation err because some fields are not here, we can add conditions to ApiDataTransformer but it's not needed, we don't need this action at all here when we leave the channel